src/storage/s3.rs

```diff
@@ -51,19 +51,25 @@ impl<'a> S3Backend<'a> {
         b.read_to_end(&mut content).unwrap();
 
         let date_updated = parse_timespec(&res.last_modified.unwrap())?;
+        let compressed = res.metadata.unwrap_or_default().get("compressed").is_some();
 
         Ok(Blob {
             path: path.into(),
             mime: res.content_type.unwrap(),
             date_updated,
             content,
+            compressed,
         })
     }
 
     pub(super) fn store_batch(&mut self, batch: &[Blob]) -> Result<(), Error> {
         use futures::stream::FuturesUnordered;
         use futures::stream::Stream;
+        use std::collections::HashMap;
+
         let mut attempts = 0;
+        let mut compressed = HashMap::with_capacity(1);
+        compressed.insert("compressed".into(), "true".into());
```

Comment:
Rather than just compressed or not, this should store the encoding used, so that it’s possible to move to different encodings in the future.

Reply:
In fact, it might make a lot of sense to use the `Content-Encoding` header instead if that’s something that S3 retains.

Reply:
`Content-Encoding` is a great idea, I didn't know it existed!

Reply:
The `rusoto_s3` field is this: https://docs.rs/rusoto_s3/0.43.0/rusoto_s3/struct.PutObjectRequest.html#structfield.content_encoding

Reply:
Added in https://github.com/rust-lang/docs.rs/pull/780/commits/f13a08783b97d6af93e6603e1a8298d1168af09f